WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,132 @@
 <% provide(:title, 'Competitions') %>
 
 <div class="container">
-  <h1><%= yield(:title) %></h1>
-
-  <%= link_to "New competition", new_competition_path %>
-  <table class="table">
-    <% @competitions.each do |competition| %>
-      <tr>
-        <td><%= competition.id %></td>
-        <td><%= competition.name %></td>
-        <td><%= link_to "Edit", admin_edit_competition_path(competition) %></td>
-        <td><%= link_to "Results", "/results/competition.php?competitionId=#{competition.id}" %></td>
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <%= form_tag competitions_path, method: :get, class: "competition-select form-inline" do %>
+    <% if @competitions.length == 0 %>
+      <%= render 'index_form' %>
+      <div class="alert alert-warning">No competitions found!</div>
+    <% else %>
+      <% if params[:display] == "List" %>
+        <table class="table competitions-table table-striped table-hover table-condensed floatThead">
+          <thead>
+            <tr>
+              <th colspan="7">
+                <%= render 'index_form' %>
+              </th>
+            </tr>
+            <tr>
+              <th>Year</th>
+              <th>Date</th>
+              <th>Name</th>
+              <th></th>
+              <th>Country, City</th>
+              <th>Venue</th>
+              <th class="big-column"></th>
+            </tr>
+          </thead>
+          <tbody>
+            <%= render 'index_table', competitions: @future_competitions %>
+            <tr id="today-line">
+              <td class="text-center"><a id="today"></a>Today</td>
+              <td colspan="6" ></td>
+            </tr>
+            <%= render 'index_table', competitions: @past_competitions %>
+          </tbody>
+        </table>
+      <% else %>
+        <%= render 'index_form' %>
+        <div id="map">
+        </div>
+      <% end %>
     <% end %>
-  </table>
+  <% end %>
 </div>
+
+<script>
+$(function() {
+  var $today = document.getElementById("today");
+  if ($today) {
+    var elOffset = $("#today").offset().top;
+    var windowHeight = $(window).height();
+    var offset = elOffset - Math.max((windowHeight - $("#today").height()) / 2, 0);
+
+    $('html, body').animate({ scrollTop: offset}, 50);
+  }
+
+  function initMap() {
+    var $map = new google.maps.Map(document.getElementById('map'), {
+      zoom: 2,
+      center: {lat: 0, lng: 0},
+      scrollwheel: true
+    });
+
+    wca.setCompetitions(<%= @competitions.map do |c|
+    {
+      id: c.id,
+      name: c.name,
+      latitude_degrees: c.latitude_degrees,
+      longitude_degrees: c.longitude_degrees,
+      start_date: c.start_date,
+      end_date: c.end_date,
+      year: c.year,
+      month: c.month,
+      day: c.day,
+      cityName: c.cityName,
+      marker_date: c.start_date.to_formatted_s(:long)
+    }
+      end.to_json.html_safe %>, $map);
+  }
+
+  wca.setCompetitions = function(competitions, map) {
```

Comment:
This seems unecessarily complicated since it will only get called once when the page loads. Can't you just iterate over `competitions`?

I'm also not comfortable with the name of this method being the same as the name of the method for the edit competitions page. There's no need to put this in the `wca` object, right? The only reason we were putting it in the `wca` object on the competitions edit page was because we needed to call that method from other files, but AFAIK, that's not the case here.